### ISSUE TITLE
Logging timestamp format

### ISSUE THREAD

**Issue by benclaussen** [PowershellFrameworkCollective/psframework#391]
I don't see any obvious way to change the format of the timestamp included in logs... Is this possible? Or perhaps possible to add? I was thinking another option such as `PSFramework.message.style.timestampformat` defaulting to the current `yyyy/MM/dd HH:mm:ss` using the [custom format strings](https://docs.microsoft.com/en-us/dotnet/standard/base-types/custom-date-and-time-format-strings)

**Comment by FriedrichWeinmann** [comment]
Heya, it's not implemented yet, but definitely something I could add with little effort.
Consider it on the list of things to include in the next release touching on logging

**Comment by benclaussen** [comment]
> 
> 
> Heya, it's not implemented yet, but definitely something I could add with little effort.
> Consider it on the list of things to include in the next release touching on logging

Awesome! Thanks for the fast response!

**Comment by FriedrichWeinmann** [comment]
Implemented and released in 1.1.59

Relevant Settings:

```
PSFramework.Logging.FileSystem.TimeFormat
PSFramework.Logging.LogFile.TimeFormat
```

Relevant parameter on Set-PSFLoggingProvider: -TimeFormat

Values default to current locale

**Comment by antoinegenot** [comment]
I'm trying to use the new "TimeFormat" paramater to add milliseconds in my log files.
I have the following command line :
`Set-PSFLoggingProvider -Name logfile -Enabled $true -FilePath $LogFilePath -TimeFormat "yyyy/MM/dd HH:mm:ss.fff"`

And the following error during the execution (sorry it is in french) :
> Le terme «PSFramework.Logging.LogFile.TimeFormat.» n'est pas reconnu comme nom d'applet de 
> commande, fonction, fichier de script ou programme exécutable. Vérifiez l'orthographe du nom, ou si un chemin d'accès existe, vérifiez que le chemin d'accès est correct et réessayez.
> Au caractère Ligne:1 : 1
> + .\MyScript.ps1
> + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : ObjectNotFound: (PSFramework.Logging.LogFile.TimeFormat.:String) [MyScript.ps1], CommandNotFoundException
>     + FullyQualifiedErrorId : CommandNotFoundException,MyScript.ps1

PSFramework module information :
- ModuleType: Script
- Version: 1.1.59
- Name: PSFramework

**Comment by antoinegenot** [comment]
I found the solution, it is not a parameter to pass directly in Set-PSFLoggingProvider cmdlet.
But this is a config for the tool, we can set it like this :
`Set-PSFConfig -FullName PSFramework.Logging.LogFile.TimeFormat "dd/MM/yyyy HH:mm:ss.fff"`

**Comment by FriedrichWeinmann** [comment]
Heya,
thanks for finding the bug and a workaround (should have worked!).
The new version of PSFramework - `1.4.149` as of this writing - should have that fixed with lots of additional goodies:
https://psframework.org/documentation/documents/psframework/logging.html

**Comment by Zerowalker** [comment]
I can't seem to change the timestamps with this, and not sure about the documentation,
where am i supposed to specify the format if not with `Set-PSFConfig`?

**Comment by chadbaldwin** [comment]
Just chiming in. I'm a bit confused as well. I just started playing with PSFramework today, but I can't seem to get the time format to change.

I'm running version `1.10.318`, just installed today for the first time.

This is the command I'm running to test...

```powershell
PS> Get-PSFConfig -Name '*.TimeFormat' | Set-PSFConfig -Value 'yyyy-MM-dd HH:mm:ss.fff'
PS> Get-PSFConfig -Name '*.TimeFormat'

FullName                                         Value
--------                                         -----
PSFramework.Logging.AzureLogAnalytics.TimeFormat yyyy-MM-dd HH:mm:ss.fff
PSFramework.Logging.EventLog.TimeFormat          yyyy-MM-dd HH:mm:ss.fff
PSFramework.Logging.FileSystem.TimeFormat        yyyy-MM-dd HH:mm:ss.fff
PSFramework.Logging.LogFile.TimeFormat           yyyy-MM-dd HH:mm:ss.fff

PS> Write-PSFMessage 'whatever' -Verbose
VERBOSE: [02:46:11][<ScriptBlock>] whatever
```

And then I check the default log file located in `AppData\Roaming\...` log file, and I see:

```
"MyPCName","2023-11-15 2:46:11 AM","Verbose","whatever"...
```

Since I'm new to this module, I can't tell if I'm just missing something, or if this is a bug based on this issue.

**Comment by FriedrichWeinmann** [comment]
Sorry about the confusion there.
For the FileSystem provider, you'll need to enable the `ModernLog` setting as well right now - without that it won't honor the TimeFormat setting ... for now.
Adding a new issue for that to fix that in the next release.

```powershell
Set-PSFConfig -FullName PSFramework.Logging.FileSystem.ModernLog -Value $true
```

This will also add headers to the debug log (which are the reason this is not the default setting, as it would be a breaking change for anybody automatically processing it).

> PSFramework.Logging.LogFile.UTC

That is a setting for the `logfile` provider, which is a different logging option than the `FileSystem` provider.
Basically, the enabled-by-default `FileSystem` logging plugin exists so you have an automatic debug log without having to do anything.
The other plugins - such as `logfile` - exist so you can have finely controlled and configured logging as you want it, if you want to actively decide on your logging. [Here's how to set that one up](https://psframework.org/documentation/documents/psframework/logging/loggingto/logfile.html)
